# March 2022

## New Py Scripts

### Notes added to the NST_notes.md file

- apps/toonify/src/NST_train_single_image.py >> Run this file to train NST on a new style and input pair. At least 2000 epochs are recommended.

- apps/toonify/src/stylization_script.py >> Run this file to transfer input image into pre-trained styles. There are 4 pre-trained models are available in models/binaries folder.

- apps/toonify/src/training_script.py >> To train NST on new style. This will save the pre-trained model which can be used to transfer images afterwards.

- apps/toonify/src/utils/ >> This utils folder consists of scripts for download pre-trained models from dropbox and other helper functions.

## Previous use of the NST script

The script was called like this:

python apps/toonify/src/NST.py

```py
parser.add_argument(
    '--input_dir', default='C:/Users/timof/repos/timofeysie/satisfactory/apps/toonify/src/test_img/Trump-Kim_Summit_June_2018_v4.jpg')
parser.add_argument(
    '--style_image', default='C:/Users/timof/repos/timofeysie/satisfactory/apps/toonify/src/style_img/Overlooking_Stampa_Augusto_Giacometti_(1908).jpg')
...
save_image(generated_image, "gen_" + opt.input_dir)
```

Here is the list of [commits](https://github.com/timofeysie/satisfactory/commits/develop) to figure out how to use the scripts.

### Scripts added

Scripts and commit messages:

```txt
NST_train_single_image.py - NST new pretrained models and training code
stylization_script.py - NST new pretrained models and training code
test.py - NST pretrained models and training code
test_image_cyclegan.py - Cyclegan pretrained models and test script
training_script.py - NST new pretrained models and training code
```

In test.py, paths were changed to relative paths:

parser.add_argument('--input_dir', default = './test_img/')

However, the generate images does not seem to be working anymore.

I have seen some strange images in the apps\toonify\src\cartooned_img directory.

Why are these files there?

Alex_De_Minaur_%2828938823637%29_sezanne.jpg
Alex_De_Minaur_%2843875330731%29_sezanne.jpg
Bundesarchiv_Bild_146-1991-014-09%2C_Stein-Pfalz%2C_Ritter_und_Justin_bei_Blutabnahme_sezanne.jpg

I think these must be the failed images El was talking about.  Was that using the Cezanne model.  Since his French name has an accent over the e: Paul CÃ©zanne, I'm guessing that's what that encoded character is: %29_sezanne

When I run the test.py script from the command line, I get this:

```txt
> python apps/toonify/src/test.py --style Hosoda --gpu 0
Traceback (most recent call last):
  File "C:\Users\timof\repos\timofeysie\satisfactory\apps\toonify\src\test.py", line 27, in <module>
    model.load_state_dict(torch.load(os.path.join(opt.model_path, opt.style + '_net_G_float.pth')), strict=False)
  File "C:\Users\timof\AppData\Local\Programs\Python\Python39\lib\site-packages\torch\serialization.py", line 594, in load   
    with _open_file_like(f, 'rb') as opened_file:
  File "C:\Users\timof\AppData\Local\Programs\Python\Python39\lib\site-packages\torch\serialization.py", line 230, in _open_file_like
    return _open_file(name_or_buffer, mode)
  File "C:\Users\timof\AppData\Local\Programs\Python\Python39\lib\site-packages\torch\serialization.py", line 211, in __init__
    super(_open_file, self).__init__(open(name, mode))
FileNotFoundError: [Errno 2] No such file or directory: './pretrained_model/Hosoda_net_G_float.pth'
PS C:\Users\timof\repos\timofeysie\satisfactory>
```

Change the new version:

parser.add_argument('--input_dir', default = './test_img/')

Back to the old version:

parser.add_argument('--input_dir', default='apps/toonify/src/test_img/')

And it works.
